Automattic/jetpack
# Social: Disable sharing UI for invalid connections

## Proposed changes:

* Disable reshare button and hide post publish share status when all the enabled connections are invalid


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Add a Facebook and an Instagram connection
* Edit an existing post
* Open Jetpack/Social sidebar and enable SIG
* Enable only the Instagram connection
* Now disable SIG to make the Instagram connection invalid due to no media
* Confirm that the "Share post" button is disabled
* Now enable any other connection
* Confirm that the button is enabled again
* Go to https://www.facebook.com/settings/?tab=business_tools and remove Jetpack to make the connection broken
* Create a new post and disable all other valid connections.
* Publish the post
* On the post publish panel, confirm that you don't see "Sharing your post..."
* Now create another post with other connections enabled
* Confirm that the share status show up fine.
* Edit an existing post
* Open Jetpack/Social sidebar
* Quickly enable broken Facebook and Instagram connections before the test results finish
* Wait for connection test results to finish
* Confirm that "Share post" button is now disabled because only the invalid connections are enabled.
* Now enable any other connection
* Confirm that the "Share post" button is enabled again
